lib/rules/no-package-self-import.js

```diff
@@ -34,12 +34,16 @@ module.exports = {
     if (!filename) return {};
 
     const packageJsonPath = findClosestPackageJsonPath(path.dirname(filename));
-
     const packageName = require(packageJsonPath).name;
 
+    /**
+     * Only flag exact matches (importPath === packageName).
+     */
     return {
       ImportDeclaration(node) {
-        if (node.source.value?.startsWith(packageName)) {
+        const importPath = node.source.value;
+        if (typeof importPath !== "string") return;
+        if (importPath == packageName) {
```

Comment:
We still need this second check to support subfolder imports. You can also include the related paths in the tests.

```suggestion
        if (importPath === packageName || importPath.startsWith(`${packageName}/`)) {
```

Reply:
This has been fixed already